src/Sylius/Component/Addressing/Model/Country.php

```diff
@@ -160,6 +160,22 @@ public function hasProvince(ProvinceInterface $province)
     /**
      * {@inheritdoc}
      */
+    public function isEqualTo($data)
+    {
+        if (is_object($data)) {
+            if (!$data instanceof CountryInterface) {
+                return false;
+            }
+
+            return $data->getId() === $this->id;
+        }
+
+        return $data === $this->id;
```

Comment:
Cause some stuff saves just ID of the object, i.e. `ShippingCountryRuleChecker`...